client/hotrod-client/src/main/java/org/infinispan/client/hotrod/telemetry/impl/TelemetryService.java

```diff
@@ -0,0 +1,25 @@
+package org.infinispan.client.hotrod.telemetry.impl;
+
+import java.nio.charset.StandardCharsets;
+
+import org.infinispan.client.hotrod.impl.protocol.HeaderParams;
+
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.Context;
+
+public class TelemetryService {
```

Comment:
Are you sure this won't cause `ClassNotFoundException` if the OTel API is not present on the classpath ?
Maybe turn `TelemetryService` into an interface and have a single implementing class `OpenTelemetryService`.